ecommerce/extensions/payment/processors/hyperpay.py

```diff
@@ -0,0 +1,226 @@
+
+        try:
+            response = requests.post(
+                checkouts_api_url,
+                request_data,
+                headers=self.authentication_headers
+            )
+            response.raise_for_status()
+            data = response.json()
+            result_code = data['result']['code']
+            if result_code != self.RESULT_CODE_SUCCESSFULLY_CREATED_CHECKOUT:
+                raise HyperPayException(
+                    'Error creating checkout. HyperPay response status code: {}'.format(result_code)
+                )
+            return data['id']
+        except requests.exceptions.RequestException as exc:
+            raise HyperPayException('Error creating a checkout due to {}'.format(exc))
+
+    def get_transaction_parameters(self, basket, request=None, use_client_side_checkout=False, **kwargs):
+        """
+        Return the transaction parameters needed for this processor.
+        """
+        payment_widget_js_url = '{}?{}'.format(
+            self.hyper_pay_api_base_url + self.PAYMENT_WIDGET_JS_PATH,
+            urlencode({'checkoutId': self._get_checkout_id(request, basket)})
+        )
+        transaction_parameters = {
+            'payment_widget_js': payment_widget_js_url,
+            'payment_page_url': reverse('hyperpay:payment-form'),
+            'payment_result_url': self.return_url,
+            'collect_billing_details_client_side': self.collect_billing_details_client_side,
+            'brands': self.BRANDS,
+            'payment_mode': self.PAYMENT_MODE,
+            'csrfmiddlewaretoken': get_token(request),
+        }
+        return transaction_parameters
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Handle the payment processor response and record the relevant details.
+        """
+        currency = response.get('currency')
+        total = response.get('amount')
+        transaction_id = response.get('id')
+        card_data = response.get('card')
+        if card_data is None:
+            card_number = 'XXXXXXXXXXXXXXXX'
+        else:
+            card_first_6_digits = card_data.get('bin', 'XXXXXX')
+            card_last_4_digits = card_data.get('last4Digits', 'XXXX')
+            card_number = '{}XXXXXX{}'.format(card_first_6_digits, card_last_4_digits)
+        payment_type = response.get('paymentBrand', 'Unknown')
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=card_number,
+            card_type=payment_type
+        )
+
+    def issue_credit(self, order_number, basket, reference_number, amount, currency):
+        """
+        This is currently not implemented.
+        """
+        logger.exception(
+            'HyperPay processor cannot issue credits or refunds from Open edX ecommerce.'
+        )
+
+
+class HyperPayMada(HyperPay):
```

Comment:
This works exactly the same as the other backend? 

Isn't it better to encapsulate the response without CC numbers on this instead of in the `HyperPay` using a if clause ([here](https://github.com/open-craft/ecommerce/pull/21/files#diff-aa3d774a9b6ef76e3ca30e5da979c4e871839a7f3d1a79219a0954a72e89489cR194-R199))?

Reply:
Nudge ^

Reply:
These are the lines I'm referring to: https://github.com/open-craft/ecommerce/pull/21/files#diff-aa3d774a9b6ef76e3ca30e5da979c4e871839a7f3d1a79219a0954a72e89489cR163-R169